### PR TITLE
7373 solr upgrade

### LINKED ISSUE #7373: Upgrade to Solr 8+
In present [Installation Guide] (https://guides.dataverse.org/en/latest/installation/prerequisites.html#supported-versions) you recommend Solr version 7, for which last update was 7.7.3, and is now only updating from version 8.x. Evaluate risk for obsolecense / end-of-life of version 7.

### PR DESCRIPTION
**Pull request by sekmiller**
**What this PR does / why we need it**:
Revises the doc and adds release notes related to upgrading Solr

**Which issue(s) this PR closes**:

Closes #7373

**Special notes for your reviewer**:
No code to review. I followed the instructions for installing Solr while using the latest stable release (8.8.1). Normal processing of indexes including reindex all for my small local db worked without a hitch. Verify that I caught all the places where the doc and install scripts need to be updated.

**Suggestions on how to test this**:
Install and run the new version of solr. verify that indexing works as expected

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No

**Is there a release notes update needed for this change?**:
doc/release-notes/7373-solr-upgrade.md

**Additional documentation**:
Doc changes were made to prerequisites, dev environment, and metadata sections

### CODE REVIEW

File: conf/docker-aio/0prep_deps.sh [hunk 1]
```diff
@@ -10,12 +10,11 @@ if [ ! -e dv/deps/payara-5.2020.6.zip ]; then
...
 
-if [ ! -e dv/deps/solr-7.7.2dv.tgz ]; then
+if [ ! -e dv/deps/solr-8.8.1dv.tgz ]; then
 	echo "solr dependency prep"
 	# schema changes *should* be the only ones...
-	cd dv/deps/
-	#wget https://archive.apache.org/dist/lucene/solr/7.3.0/solr-7.3.0.tgz -O solr-7.3.0dv.tgz
-	wget https://archive.apache.org/dist/lucene/solr/7.7.2/solr-7.7.2.tgz -O solr-7.7.2dv.tgz
+	cd dv/deps/	
+	wget https://archive.apache.org/dist/lucene/solr/8.8.2/solr-8.8.1.tgz -O solr-8.8.1dv.tgz
```
Comment (scolapasta) [inline conf/docker-aio/0prep_deps.sh:17]: 8.8.2 directory, but 8.81 version?

File: doc/release-notes/7373-solr-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+### Solr Update
+
+With this release we upgrade to the latest available stable release in the Solr 8.x branch. We recommend a fresh installation of Solr 8.8.1 (the index will be empty)
```
Comment (scolapasta) [inline doc/release-notes/7373-solr-upgrade.md:3]: Do we want to consider porting the old index and incremental index all? (what have we done for past Solr upgrades?)

**Review by scolapasta — COMMENTED** [review]
Looks good - just some questions to consider.

File: conf/docker-aio/0prep_deps.sh (hunk 1 above)
Comment (sekmiller) [reply conf/docker-aio/0prep_deps.sh:17]: Typo. will fix.

File: doc/release-notes/7373-solr-upgrade.md (hunk 2 above)
Comment (sekmiller) [reply doc/release-notes/7373-solr-upgrade.md:3]: I didn't consider that. This is close to the release note we used when we went to 7.7.2

File: scripts/vagrant/setup-solr.sh [hunk 3]
```diff
@@ -5,12 +5,12 @@ SOLR_USER=solr
...
 mkdir $SOLR_HOME
 chown $SOLR_USER:$SOLR_USER $SOLR_HOME
-su $SOLR_USER -s /bin/sh -c "cp /dataverse/downloads/solr-7.7.2.tgz $SOLR_HOME"
-su $SOLR_USER -s /bin/sh -c "cd $SOLR_HOME && tar xfz solr-7.7.2.tgz"
-su $SOLR_USER -s /bin/sh -c "cd $SOLR_HOME/solr-7.7.2/server/solr && cp -r configsets/_default . && mv _default collection1"
-su $SOLR_USER -s /bin/sh -c "cp /dataverse/conf/solr/7.7.2/schema*.xml $SOLR_HOME/solr-7.7.2/server/solr/collection1/conf/"
-su $SOLR_USER -s /bin/sh -c "cp /dataverse/conf/solr/7.7.2/solrconfig.xml $SOLR_HOME/solr-7.7.2/server/solr/collection1/conf/solrconfig.xml"
-su $SOLR_USER -s /bin/sh -c "cd $SOLR_HOME/solr-7.7.2 && bin/solr start && bin/solr create_core -c collection1 -d server/solr/collection1/conf/"
+su $SOLR_USER -s /bin/sh -c "cp /dataverse/downloads/solr-8.8.1.tgz $SOLR_HOME"
+su $SOLR_USER -s /bin/sh -c "cd $SOLR_HOME && tar xfz solr-78.8.1.tgz"
```
Comment (pdurbin) [inline scripts/vagrant/setup-solr.sh:9]: Is this right? 78.8.1?

**Review by pdurbin — COMMENTED** [review]
I only took a quick look but I think I found a typo in the Vagrant script.

Overall, I'm glad that both Vagrant and docker-aio were updated.

**Review by pdurbin — COMMENTED** [review]
Should we update the following in pom.xml?

```
<dependency>
    <groupId>org.apache.solr</groupId>
    <artifactId>solr-solrj</artifactId>
    <version>7.7.2</version>
</dependency>
```

**Comment by djbrooke** [comment]
Thanks @pdurbin @scolapasta for the reviews (and @sekmiller for the PR). Is there anything left here or can we smash the approve button?

**Review by scolapasta — APPROVED** [review]
OK, after discussing at tech hours, we decided we're ok with the release notes, as is.

**Review by pdurbin — COMMENTED** [review]
Sorry, I pulled this out of QA because I thought of something.

Should we delete `conf/solr/7.7.2` and all of its contents? That's what we usually do. Otherwise we have two directories lying around (7.7.2 and 8.8.1) which could lead to confusion.

**Comment by pdurbin** [comment]
> Should we delete `conf/solr/7.7.2` and all of its contents?

Fixed in ed6f575. Moving back to QA.

**Review by pdurbin — APPROVED** [review]

**Comment by kcondon** [comment]
Issues found so far:
[x] 1. add how to handle custom metadata blocks to release notes
Stephen add note for additional steps, pointing to metadata blocks section of manual and update script as in release notes from 5.1
2. make script complains about schema.xml
```
copying jhove schema file
/bin/cp ../../conf/jhove/jhoveConfig.xsd dvinstall
make: *** No rule to make target `../../conf/solr/7.7.2/schema.xml', needed by `dvinstall/schema.xml'.  Stop.
```

**Pull request merged**